Sources/DeepLearning/Optimizer.swift

```diff
@@ -16,13 +16,16 @@
 import TensorFlow
 #endif
 
+public protocol Optimizable: Differentiable & KeyPathIterable
```

Comment:
Thanks for introducing this change! However, empty protocols that are only used for joining constraints are generally discouraged in Swift. Could you revert it so that we won't introduce any API breakage in this PR?

Reply:
Sounds good! Is it ok if I keep the change of `Model: Layer` -> `Model: Differentiable & KeyPathIterable` (with the where clause) for the optimizer implementations in the file? This is so they can be used for more than just layers.